staging/src/k8s.io/kubectl/pkg/util/prune/prune.go

```diff
@@ -35,26 +54,15 @@ func (pr Resource) String() string {
 	return fmt.Sprintf("%v/%v, Kind=%v, Namespaced=%v", pr.group, pr.version, pr.kind, pr.namespaced)
 }
 
-func GetRESTMappings(mapper meta.RESTMapper, pruneResources []Resource) (namespaced, nonNamespaced []*meta.RESTMapping, err error) {
+// if namespace specified, the default allow list would not include namespace and persistent volume.
+// if pruneResources is specified by user, respect the user setting.
+func GetRESTMappings(mapper meta.RESTMapper, pruneResources []Resource, namespaceSpecified bool) (namespaced, nonNamespaced []*meta.RESTMapping, err error) {
 	if len(pruneResources) == 0 {
-		// default allowlist
-		pruneResources = []Resource{
-			{"", "v1", "ConfigMap", true},
-			{"", "v1", "Endpoints", true},
-			{"", "v1", "Namespace", false},
-			{"", "v1", "PersistentVolumeClaim", true},
-			{"", "v1", "PersistentVolume", false},
-			{"", "v1", "Pod", true},
-			{"", "v1", "ReplicationController", true},
-			{"", "v1", "Secret", true},
-			{"", "v1", "Service", true},
-			{"batch", "v1", "Job", true},
-			{"batch", "v1", "CronJob", true},
-			{"networking.k8s.io", "v1", "Ingress", true},
-			{"apps", "v1", "DaemonSet", true},
-			{"apps", "v1", "Deployment", true},
-			{"apps", "v1", "ReplicaSet", true},
-			{"apps", "v1", "StatefulSet", true},
+		pruneResources = defaultPruneResources
+		// TODO in future release, add back non-namespaced resource only if namespace is not specified
+		pruneResources = append(pruneResources, Resource{"", "v1", "Namespace", false}, Resource{"", "v1", "PersistentVolume", false})
+		if !namespaceSpecified {
```

Comment:
Don't we want to print this when namespace IS specified, because that's the case where the behaviour will change? Can we test for when this is/isn't printed please?

Reply:
I make it wrongly.